gtk4/src/subclass/widget.rs

```diff
@@ -876,6 +880,82 @@ pub unsafe trait WidgetClassSubclassExt: ClassStruct {
         }
     }
 
+    #[doc(alias = "gtk_widget_class_install_action")]
+    fn install_action<
+        F: Fn(&Widget, &str, Option<&Variant>) + 'static,
+        S: WidgetImpl + ObjectSubclass,
```

Comment:
I'm not sure whether i need something similar here, or if there's a better way to pass Self as S to trampoline's scope?

Reply:
You could require that `Self: ClassStruct` and then go via that. See the impl line further below `unsafe impl<T: ClassStruct> WidgetClassSubclassExt for T where T::Type: WidgetImpl {}`.